### DISCUSSION TITLE
Docking : how to open imgui window?

### DISCUSSION THREAD

**Discussion by sy-project** [ocornut/imgui#7005] [New Users Build/Link/Run issues ONLY!]
Version/Branch of Dear ImGui
```
Version: 1.90 WIP (18995)
Branch : Docking
```

My Issue/Question:
I cant open imgui window in dx11.
what could be the problem?

My code is here
```
https://github.com/sy-project/yuuna
```
class "cSystem, Camera, Environment" called ImGui.

and only ImGui Demo window can active.
![image](https://github.com/ocornut/imgui/assets/42532250/67e03e38-622c-4cdd-aa61-3ff2bd51e848)

**Comment by ocornut** [comment]
Moving you to a Discussion since you did not read nor fulfill Issues submitting Guidelines.
We cannot help you if we don't have sufficient details on what you did.

**Comment by sy-project** [comment]
ImGui Initialize Code is here
```
    IMGUI_CHECKVERSION();

    ImGui::CreateContext();

    ImGuiIO& io = ImGui::GetIO(); (void)io;
    io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;
    io.ConfigFlags |= ImGuiConfigFlags_DockingEnable;
    io.ConfigFlags |= ImGuiConfigFlags_ViewportsEnable;

    ImGui::StyleColorsDark();

    ImGuiStyle& imgui_style = ImGui::GetStyle();
    if (io.ConfigFlags & ImGuiConfigFlags_ViewportsEnable)
    {
        imgui_style.WindowRounding = 0.0f;
        imgui_style.Colors[ImGuiCol_WindowBg].w = 1.0f;
    }

    ImGui_ImplWin32_Init(g_hWnd);
    ImGui_ImplDX11_Init(Device::Get()->GetDevice(), Device::Get()->GetDeviceContext());
```
And I tried this code. but I cant see the Test Window.
```
    ImGui_ImplDX11_NewFrame();
    ImGui_ImplWin32_NewFrame();
    ImGui::NewFrame();
    static bool show_test = true;
    if (show_test) {
        if (ImGui::Begin("Test", &show_test, 1024)) {
            ImGui::Text("Test");
        }
        ImGui::End();
    }
    ImGui::ShowDemoWindow(&show_demo_window);
    Environment::Get()->PostRender();
    ImGui::Render();
    ImGui_ImplDX11_RenderDrawData(ImGui::GetDrawData());
    ImGui::UpdatePlatformWindows();
    ImGui::RenderPlatformWindowsDefault();
```

**Comment by GamingMinds-DanielC** [comment]
From the Demo Window menu bar, select "Tools" -> "Metrics/Debugger". In that window expand "Windows" and take a closer look. Maybe your window is at a position where you cannot see it, in case you configured it in a strange way first and then those settings got saved. Deleting your imgui.ini might remedy that. If your window isn't listed there, you need to step through your code to find out why it doesn't get submitted.

Btw, it's a bad idea to use direct values like "1024" for flags, it just makes code harder to read. and error prone in case those values ever change in a new version. You can and should use the named enum values directly, `ImGuiWindowFlags_MenuBar` in this case.
